### PR TITLE
Update UkrlpService to v6

### PR DESCRIPTION
**Pull request by timabell**
See individual commit messages for detailed changes.

Here's a representative example of how the xml structure changed between the versions:

address field mapping with county example

[PTCD-630](https://skillsfundingagency.atlassian.net/browse/PTCD-630)

### CODE REVIEW

File: tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs [hunk 1]
```diff
@@ -19,123 +24,111 @@ public UkrlpSyncHelperTests(DatabaseTestBaseFixture fixture)
         }
 
         [Fact]
-        public async Task SyncProviderData_ProviderAlreadyExists_UpdatesProviderInfo()
+        public async Task SyncProviderData_CreatesProvider()
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs:27]: Nit: most of our tests follow the `MethodName_StateUnderTest_ExpectedBehavior` naming convention so previous naming would be better in this regard

File: tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs [hunk 2]
```diff
@@ -19,123 +24,111 @@ public UkrlpSyncHelperTests(DatabaseTestBaseFixture fixture)
...
+            createCommand.ProviderStatus.Should().Be(ukrlpData.ProviderStatus);
+            createCommand.ProviderType.Should().Be(ProviderType.Both);
+            createCommand.Status.Should().Be(ProviderStatus.Registered);
+            createCommand.Ukprn.Should().Be(ukprn);
+            var actualContact = createCommand.Contacts.Should().ContainSingle().Subject;
+            AssertContactMapping(actualContact, ukrlpContact);
+        }
+
+        [Fact]
+        public async Task SyncProviderData_UpdatesExistingProvider()
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs:55]: As above

**Review by gunndabad — APPROVED** [review]

File: tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs (hunk 1 above)
Comment (timabell) [reply tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs]: Done :+1:

File: tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs (hunk 2 above)
Comment (timabell) [reply tests/Dfc.CourseDirectory.Core.Tests/ReferenceDataTests/UkrlpSyncHelperTests.cs]: Done :+1:

**Comment by timabell** [comment]
Touched last commit and forcepushed to trigger a new CI build. We seem to have package restore issues on Azure DevOops

**Comment by timabell** [comment]
[Release for testing](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_releaseProgress?releaseId=16192&_a=release-pipeline-progress) (this doesn't have the last test change because the builds are borked at the moment but this has the current production code so is fine to test with)

**Comment by timabell** [comment]
rebased to fix build

**Comment by timabell** [comment]
[Updated release for testing](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_releaseProgress?releaseId=16198&_a=release-pipeline-progress) (now that builds are fixed)

**Comment by timabell** [comment]
Issue raised in test. Possibly not pulling contact name from xml. Need to add a test that includes deserialization of actual xml sample.

**Comment by timabell** [comment]
Closing this because I'm going to be opening a new PR from a new branch.
The new branch has additional regression tests added for the v3 (before the upgrade in the commit history); so with all the history rewriting this PR doesn't quite make sense any more.

**Pull request closed without merging**